dts/riscv32/rv32m1_zero_riscy.dtsi

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2018 Foundries.io Ltd
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/interrupt-controller/openisa-intmux.h>
```

Comment:
Seems like we should have a common rv32m1.dtsi or something that is shared between rv32m1_zero_riscy.dtsi and rv32m1_ri5cy.dtsi

Reply:
The interrupts are different for each core

Reply:
How do the interrupts differ?  I remember starting to look at how the chained interrupt controller worked on this SoC.  I've got an in progress change for DTS to generate the proper Zephyr IRQ defines.

Reply:
The `interrupts` properties are different, and in some cases the `interrupt-parent` properties are also different. GPIOE, for example, goes through the EVENT unit on the ZERO-RISCY core, but through the INTMUX unit on the RI5CY core.
https://github.com/NXPmicro/zephyr/blob/f0e02034f2987d6421715e69dfb4c049173ec297/dts/riscv32/rv32m1_ri5cy.dtsi#L202-L211
https://github.com/NXPmicro/zephyr/blob/f0e02034f2987d6421715e69dfb4c049173ec297/dts/riscv32/rv32m1_zero_riscy.dtsi#L179-L187

Reply:
While the `interrupts` are different the other bits are the same.  I'd rather see we have some common .dtsi for all the details that are common and than have rv32m1_ri5cy.dtsi and rv32m1_zero_riscy.dtsi have the interrupt details that are unique.

Reply:
I thought this would still end up with a lot of code repetition, but it turned out better than I expected. Pushed a new update, please have a look.